docs/proposals/pvcs-smaller-than-1gi.md

```diff
@@ -0,0 +1,543 @@
+
+   For this step, we can keep the existing call in place and do not need to touch anything.
+
+2. >`external-provisioner` calls CSI controller's `CreateVolume` with the topology key of the target node.
+
+   For this step, we will need to adjust how `CreateVolume` passes the requested storage to allow less than 1Gi being a valid capacity.
+
+3. >`topolvm-controller` creates a `LogicalVolume` with the topology key and capacity of the volume.
+
+   For this step, we will need to make sure that `topolvm-controller` is able to handle less than 1Gi storage as a valid capacity inside `LogicalVolume`.
+
+4. > `topolvm-node` on the target node finds the `LogicalVolume`.
+
+   We do not need to touch the existing discovery logic.
+
+5. > `topolvm-node` sends a volume create request to `lvmd`.
+
+   **BREAKING**: We need to make sure that the request sent to lvmd via gRPC is able to convey capacities less than 1Gi, which is currently not possible.
+   Similarly, all responses must be able to work with less than 1Gi capacities. 
+   More details on this can be found in the section on the [Impact on LVMD protocol changes](#switching-all-occurrences-of-uint64-to-int64-in-lvmd-and-change-sizegb-to-size-in-protobuf-messages).
+
+6. > `lvmd` creates an LVM logical volume as requested.
+
+   No changes to creation are expected, since lvm2 can deal with less than 1Gi storage already.
+
+7. > `topolvm-node` updates the status of `LogicalVolume`.
+
+   No changes are expected as the `LogicalVolume` status supports reporting capacities less than 1Gi
+
+8. > `topolvm-controller` finds the updated status of `LogicalVolume`.
+
+   As long as `LogicalVolume` has the correct capacities we do not need to change anything here.
+
+9. > `topolvm-controller` sends the success (or failure) to `external-provisioner`.
+
+   No changes are expected after sizing has been completed
+
+10. > `external-provisioner` creates a PersistentVolume (PV) and binds it to the PVC.
+
+    No changes are expected after sizing has been completed
+
+
+Additionally, any TopoLVM user making use of the [scheduler extension](../topolvm-scheduler.md) will also be impacted by the following changes:
+
+1. > `topolvm-node` exposes free storage capacity as `capacity.topolvm.io/<device-class>` annotation of each Node.
+
+   These capacities are already reported on byte-level, not in Gi, so they do not need to be touched
+2. > `topolvm-controller` works as a [mutating webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) for new Pods.
+   > - It adds `capacity.topolvm.io/<device-class>` annotation to a pod and `topolvm.io/capacity` resource to the first container of a pod.
+   > - The value of the annotation is the sum of the storage capacity requests of unbound TopoLVM PVCs for each volume group referenced by the pod.
+
+   This logic is not impacted by sizing less than 1Gi and can be left untouched
```

Comment:
You are absolutely right, I missed this originally.

This logic currently rounds up to the next GB value and needs to be modified from `requested = ((req.Value()-1)>>30 + 1) << 30` to 
`requested = req.Value()` 
for the Mutator in all places.

Otherwise, the capacity annotation will be incorrect.

Reply:
>This logic currently rounds up to the next GB value and needs to be modified from
>requested = ((req.Value()-1)>>30 + 1) << 30 to
>requested = req.Value()
>for the Mutator in all places.

confirmed.

> Actually also then we also don't need the DefaultCapacity if we are on anything > 0.
Yeah. We can delete it entirely if we use byte units. IMO.